pkg/api/vm/handler.go

```diff
@@ -69,3 +93,59 @@ func (h *vmActionHandler) restartVM(name, namespace string) error {
 
 	return nil
 }
+
+func (h *vmActionHandler) ejectCdRom(name, namespace string, diskNames []string) error {
+	vm, err := h.vmCache.Get(namespace, name)
+	if err != nil {
+		return err
+	}
+
+	vmCopy := vm.DeepCopy()
+	if err := ejectCdRomFromVM(vmCopy, diskNames); err != nil {
+		return err
+	}
+
+	if !reflect.DeepEqual(vm, vmCopy) {
+		if _, err := h.vms.Update(vmCopy); err != nil {
+			return err
+		}
+	}
+
+	return h.vmis.Delete(namespace, name, &metav1.DeleteOptions{})
```

Comment:
Is this expected? The vm restarts and services are down when users try to do a cdrom ejection

Reply:
CD-ROM  ejection takes effect after restarted, we can warn user this action will restart VM when user use this function